Chess/painter.cpp

```diff
@@ -1,25 +1,105 @@
-#include "painter.h"
-
-Painter::Painter() {}
-
-void Painter::initializeWindow() {
-
-    initwindow(WINDOW_WIDTH, WINDOW_HEIGHT, "Chess");
-}
-
-void Painter::initializeBoard() {
-    // TODO
-}
-
-void Painter::initializePieces() {
-    initializeBlackPieces();
-    initializeWhitePieces();
-}
-
-void Painter::initializeBlackPieces() {
-    // TODO
-}
-
-void Painter::initializeWhitePieces() {
-    // TODO
-}
+#include "painter.h"
+
+Painter::Painter() {}
+
+void Painter::initializeWindow()
+{
+
+    initwindow(WINDOW_WIDTH, WINDOW_HEIGHT, "Chess");
+}
+
+void Painter::initializeBoard()
+{
+
+    for(int i = 1; i <= 8; i++)
+    {
+        for(int j = 1; j <= 8; j++)
+        {
+
+            if((i + j) % 2)
+                drawSquare(i, j, 0, "Backgrounds/Black.jpg");
+            else
+                drawSquare(i, j, 1, "Backgrounds/White.jpg");
+
+        }
+    }
+}
+
+void Painter::initializePieces()
+{
+    initializeBlackPieces();
+    initializeWhitePieces();
+}
+
+void Painter::initializeBlackPieces()
+{
+
+    for(int i = 1; i <= 8; i++)
+    {
+        for(int j = 1; j <= 2; j++)
+        {
+            if((i + j) % 2)
+            {
+                if(j == 1)
+                    loadPiece(i, j, BlackOnBlackPaths[i / 2 - 1]);

```

Comment:
As extra loadPiece intr-o clasa separata denumita ```PiecePainter```

Reply:
Si sa se ocupe clasa respectiva cu determinatul tipului de piesa in functie de row/column